rq/job.py

```diff
@@ -1679,8 +1695,77 @@ def __init__(self, max: int, interval: Union[int, Iterable[int]] = 0):
         self.intervals = intervals
 
 
+class CallbackType(str, Enum):
+    SUCCESS = 'success'
+    FAILURE = 'failure'
+    STOPPED = 'stopped'
+
+
+@runtime_checkable
```

Comment:
Should these really be runtime checkable? It seems like this should just be left for a type checking tool to check that it's the right type, if it's the wrong type at runtime you'll see an error when the job is invoked.